lua/weapons/weapon_dodgeball/cl_init.lua

```diff
@@ -5,7 +5,21 @@ SWEP.Instructions = "Primary: Throw hard\nSecondary: Throw soft\nReload: Pass"
 SWEP.DrawAmmo = false
 SWEP.ViewModelFOV = 85
 
+function SWEP:ThrownBallExists()
+    local ply = self:GetOwner()
+    if(IsValid(self:GetThrownBall()))then return true end
+    if(IsValid(self.fakethrownball))then return true end
+    
+    local lookent = ply:GetEyeTrace().Entity
+    if(IsValid(lookent) and lookent:GetClass() == "dodgeball" and lookent:GetPos():Distance(ply:EyePos()) < 128)then
+        self.fakethrownball = lookent
```

Comment:
so if you're carrying a dodgeball, then walk up to another dodgeball on the ground and look at it, then the one you're holding turns invisible forever?

Reply:
It should only use this value once thrown. Is there not a check?

Reply:
looks like there's not